hack/istio-config-1.10.0.yaml

```diff
@@ -7,7 +7,7 @@ spec:
   hub: gcr.io/istio-release
   meshConfig:
     # Change the following line to configure the trust domain of the Istio cluster.
-    trustDomain: foo.bar
+    trustDomain: cluster.local
```

Comment:
It's not clear to me why this is changing. We should either change back, or change every istio configuration file to this value.

Reply:
> It's not clear to me why this is changing. 

My bad, probably should've commented on this one.

The reason for this is that the current guide in the README (i.e. the one on `main`) uses `hack/istio-config-1.10.0.yaml` verbatim for its example, but also suggests installing the helm chart whose default `app.tls.trustDomain` is `cluster.local`, which is one of the foot guns that everyone hit while doing the customer empathy session; i.e. this is one of the things that means if you follow the instructions you'll end up with a cluster that doesn't work.

> We should either change back, or change every istio configuration file to this value.

Yeah, fair point! I think probably either:

1. Change this back and set `app.tls.trustDomain` to `foo.bar` in the getting started guide
2. Provide `docs/istio-config-getting-started.yaml` for use in the getting started guide.

Given that the `hack/istio-config-x.yaml` files are used in the [e2e tests / demo](https://github.com/cert-manager/istio-csr/blob/ed48ecf4d305f0f970685bde1d23481ab0dc6940/hack/demo/deploy-demo.sh#L54), I think probably we want option 2, since we don't want tweaks to the e2e tests to affect our docs or vice versa. I'll make that change.